### PR TITLE
Removed redundant null tests when sorting a secondary values array.

### PR DESCRIPTION
**Pull request by colgreen**
#14162

### CODE REVIEW

File: src/mscorlib/src/System/Collections/Generic/ArraySortHelper.cs [hunk 1]
```diff
@@ -154,8 +155,8 @@ internal static void Sort(T[] keys, int index, int length, Comparison<T> compare
 
         internal static int InternalBinarySearch(T[] array, int index, int length, T value, IComparer<T> comparer)
         {
-            Debug.Assert(array != null, "Check the arguments in the caller!");
-            Debug.Assert(index >= 0 && length >= 0 && (array.Length - index >= length), "Check the arguments in the caller!");
+            Contract.Requires(array != null, "Check the arguments in the caller!");
```
Comment (jkotas) [inline src/mscorlib/src/System/Collections/Generic/ArraySortHelper.cs]: Looks like incorrectly resolved conflict. Could you please undo these changes?

Comment (colgreen) [reply src/mscorlib/src/System/Collections/Generic/ArraySortHelper.cs]: Hmm ok, I erroneously submitted against a older version (Contract.Requires() was changed to Debug.Assert() throughout). Not sure what the process is; should I delete the pull request and submit another? (Fishing for advice)

Comment (colgreen) [reply src/mscorlib/src/System/Collections/Generic/ArraySortHelper.cs]: ok think I figured it out and fixed all the problems with the pull request.

**Comment by jkotas** [comment]
@colgreen Could you please sign the contribution license agreement at https://cla.dotnetfoundation.org/dotnet/coreclr?pullRequest=14277 so that this can be merged?

**Comment by colgreen** [comment]
@jkotas ok I have signed the CLA.

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by jkotas** [comment]
@colgreen Thanks!
